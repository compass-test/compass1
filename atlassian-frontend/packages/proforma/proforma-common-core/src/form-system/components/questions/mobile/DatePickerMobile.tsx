import React from 'react';

import { InjectedIntlProps, injectIntl } from 'react-intl';
import styled from 'styled-components';

import {
  EditDateMessages,
  IntlEditDateMessages,
} from '../edit/EditDateMessages.intl';

interface DatePickerMobileProps {
  id?: string;
  dateValue?: string;
  onChange?: (value: string) => void;
}

export const DatePickerMobile = injectIntl(
  ({
    id,
    dateValue,
    onChange,
    intl,
  }: DatePickerMobileProps & InjectedIntlProps) => {
    return (
      <DatePickerMobileStyles>
        <input
          id={id}
          type="date"
          value={dateValue}
          onChange={
            onChange ? event => onChange(event.target.value) : undefined
          }
          placeholder={intl.formatMessage(
            IntlEditDateMessages[EditDateMessages.DatePlaceholder],
            { date: intl.formatDate(new Date(), {}) },
          )}
        />
      </DatePickerMobileStyles>
    );
  },
);

const DatePickerMobileStyles = styled.div`
  & > input {
    white-space: initial;
    display: flex;
    background-color: #f4f5f7;
    border-color: #f4f5f7;
    border-radius: 3px;
    margin-bottom: 10px;
    box-shadow: none;
    border-style: solid;
    border-width: 2px;
    padding-top: 10px;
    padding-bottom: 9px;
    padding-left: 6px;
    box-sizing: border-box;
    width: 100%;

    &:hover {
      background-color: #ebecf0;
    }
  }
`;
