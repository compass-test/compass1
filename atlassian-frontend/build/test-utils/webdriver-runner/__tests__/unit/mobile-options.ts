import { getBlendedMobileOptions } from '../../lib/clients/mobile/mobile-options';

const DEFAULT_OPTIONS = getBlendedMobileOptions();

describe('MobileTestCaseOptions.getBlendedMobileOptions', () => {
  it('should return default options when no overrides exist', () => {
    const empty = getBlendedMobileOptions({});
    expect(empty).toEqual(DEFAULT_OPTIONS);
    expect(DEFAULT_OPTIONS).toMatchSnapshot();
  });

  it(`should remove appropriate platform versions when skipPlatform is used`, () => {
    const androidOnly = getBlendedMobileOptions({ skipPlatform: ['ios'] });
    expect(androidOnly.versions).toMatchSnapshot('Android only');
  });

  it(`should replace wildcard with all platforms`, () => {
    const androidOnly = getBlendedMobileOptions({ skipPlatform: ['*'] });
    expect(androidOnly.versions).toMatchSnapshot('Wildcard');
  });

  it('should use defaults when provided an empty versions array', () => {
    const replacedVersions = getBlendedMobileOptions({
      versions: [],
    });
    expect(replacedVersions.versions).toMatchSnapshot();
  });

  it('should replace default versions when a populated versions array is provided', () => {
    const replacedVersions = getBlendedMobileOptions({
      versions: ['ios 12', 'android 9'],
    });
    expect(replacedVersions.versions).toMatchSnapshot();
  });

  it(`should augment default versions when a provided versions array includes 'DEFAULT'`, () => {
    const augmentedVersions = getBlendedMobileOptions({
      versions: ['DEFAULT', 'android 9', 'ios 12'],
    });
    expect(augmentedVersions.versions).toMatchSnapshot();
  });

  it(`should replace default form factors when a formFactors array is provided`, () => {
    const formFactors = getBlendedMobileOptions({
      formFactors: ['tablet'],
    });
    expect(formFactors.formFactors).toMatchSnapshot();
  });

  it(`should use defaults when provided an empty formFactors array`, () => {
    const formFactors = getBlendedMobileOptions({
      formFactors: [],
    });
    expect(formFactors.formFactors).toMatchSnapshot();
  });

  it(`should replace default keyboards when a keyboards array is provided`, () => {
    const keyboards = getBlendedMobileOptions({
      keyboards: ['samsung'],
    });
    expect(keyboards.keyboards).toMatchSnapshot();
  });

  it(`should use defaults when provided an empty keyboards array`, () => {
    const keyboards = getBlendedMobileOptions({
      keyboards: [],
    });
    expect(keyboards.keyboards).toMatchSnapshot();
  });
});
