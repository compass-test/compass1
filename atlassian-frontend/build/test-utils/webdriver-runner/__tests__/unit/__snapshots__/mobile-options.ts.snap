// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileTestCaseOptions.getBlendedMobileOptions should augment default versions when a provided versions array includes 'DEFAULT' 1`] = `
Array [
  "android 9",
  "android 10",
  "ios 12",
  "ios 14",
]
`;

exports[`MobileTestCaseOptions.getBlendedMobileOptions should remove appropriate platform versions when skipPlatform is used: Android only 1`] = `
Array [
  "android 10",
]
`;

exports[`MobileTestCaseOptions.getBlendedMobileOptions should replace default form factors when a formFactors array is provided 1`] = `
Array [
  "tablet",
]
`;

exports[`MobileTestCaseOptions.getBlendedMobileOptions should replace default keyboards when a keyboards array is provided 1`] = `
Array [
  "samsung",
]
`;

exports[`MobileTestCaseOptions.getBlendedMobileOptions should replace default versions when a populated versions array is provided 1`] = `
Array [
  "android 9",
  "ios 12",
]
`;

exports[`MobileTestCaseOptions.getBlendedMobileOptions should replace wildcard with all platforms: Wildcard 1`] = `
Array [
  "android 10",
  "ios 14",
]
`;

exports[`MobileTestCaseOptions.getBlendedMobileOptions should return default options when no overrides exist 1`] = `
Object {
  "formFactors": Array [
    "phone",
  ],
  "keyboards": Array [
    "apple",
    "gboard",
  ],
  "skipPlatform": undefined,
  "versions": Array [
    "android 10",
    "ios 14",
  ],
}
`;

exports[`MobileTestCaseOptions.getBlendedMobileOptions should use defaults when provided an empty formFactors array 1`] = `
Array [
  "phone",
]
`;

exports[`MobileTestCaseOptions.getBlendedMobileOptions should use defaults when provided an empty keyboards array 1`] = `
Array [
  "apple",
  "gboard",
]
`;

exports[`MobileTestCaseOptions.getBlendedMobileOptions should use defaults when provided an empty versions array 1`] = `
Array [
  "android 10",
  "ios 14",
]
`;
